src/Tizen.Multimedia.MediaPlayer/Player/Player.cs

```diff
@@ -576,24 +576,23 @@ public int GetPlayPosition()
             ret.ThrowIfFailed(this, "Failed to set play position");
         }
 
+        internal static NativePlayer.SeekCompletedCallback seekCompletedCallback;
         private async Task SetPlayPosition(long position, bool accurate, bool nanoseconds)
         {
             var taskCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
             bool immediateResult = _source is MediaStreamSource;
 
-            NativePlayer.SeekCompletedCallback cb = _ => taskCompletionSource.TrySetResult(true);
+            seekCompletedCallback = _ => taskCompletionSource.TrySetResult(true);
 
-            using (var cbKeeper = ObjectKeeper.Get(cb))
+            NativeSetPlayPosition(position, accurate, nanoseconds, seekCompletedCallback);
+            if (immediateResult)
             {
-                NativeSetPlayPosition(position, accurate, nanoseconds, cb);
-                if (immediateResult)
-                {
-                    taskCompletionSource.TrySetResult(true);
-                }
-
-                await taskCompletionSource.Task;
+                taskCompletionSource.TrySetResult(true);
```

Comment:
Actually, `NativePlayer.SeekCompletedCallback cb` is not released by GC, before async method was completed even though does not use ObjectKeeper class.
All local reference in async method are keep till complete of async method.

When `_source is MediaStreamSource` cb will released by GC before called the callback , because `taskCompletionSource.TrySetResult(true);` make complete this async method.  

If you want fix this issue. 
Do not pass cb in NativeSetPlayPosition when `_source is MediaStreamSource`

Reply:
ok. that looks better. thank you for comments. i've patched it.